Civi/Token/TokenCompatSubscriber.php

```diff
@@ -131,8 +131,16 @@ public function onRender(TokenRenderEvent $e) {
 
     if ($useSmarty) {
       $smarty = \CRM_Core_Smarty::singleton();
+      $handler = set_error_handler([$this, 'handleSmartyError'], E_USER_ERROR);
       $e->string = $smarty->fetch("string:" . $e->string);
+      set_error_handler($handler);
```

Comment:
I think the way to reset is to use restore_error_handler(): https://www.php.net/manual/en/function.restore-error-handler.php

Reply:
good spotting - fixed